scipy/stats/_continuous_distns.py

```diff
@@ -9309,6 +9309,200 @@ def _updated_ctor_param(self):
+        cython_symbol = '_studentized_range_moment'
+        _a, _b = self._get_support()
+        # all three of these are used to create a numpy array so they must
+        # be the same shape.
+
+        def _single_moment(K, k, df):
+            log_const = _stats._studentized_range_pdf_logconst(k, df)
+            arg = [K, k, df, log_const]
+            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(_stats, cython_symbol, usr_data)
+
+            ranges = [(-np.inf, np.inf), (0, np.inf), (_a, _b)]
+            opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_moment, 3, 1)
+        return np.float64(ufunc(K, k, df))
+
+    def _pdf(self, x, k, df):
+        cython_symbol = '_studentized_range_pdf'
+
+        def _single_pdf(q, k, df):
+            log_const = _stats._studentized_range_pdf_logconst(k, df)
+            arg = [q, k, df, log_const]
+            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(_stats, cython_symbol, usr_data)
+
+            ranges = [(-np.inf, np.inf), (0, np.inf)]
+            opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_pdf, 3, 1)
+        return np.float64(ufunc(x, k, df))
+
+    def _cdf(self, x, k, df):
+
+        def _single_cdf(q, k, df):
+            # "When the degrees of freedom V are infinite the probability
+            # integral takes [on a] simpler form," and a single asymptotic
+            # integral is evaluated rather than the standard double integral.
+            # (Lund, Lund, page 205)
+            if df < 100000:
+                cython_symbol = '_studentized_range_cdf'
+                log_const = _stats._studentized_range_cdf_logconst(k, df)
+
+                arg = [q, k, df, log_const]
+                usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+                llc = LowLevelCallable.from_cython(
+                    _stats, cython_symbol, usr_data)
+
+                ranges = [(-np.inf, np.inf), (0, np.inf)]
+                opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+            else:
+                cython_symbol = '_studentized_range_cdf_asymptotic'
+
+                arg = [q, k]
+                usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+                llc = LowLevelCallable.from_cython(
+                    _stats, cython_symbol, usr_data)
+
+                ranges = [(-np.inf, np.inf)]
+                opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_cdf, 3, 1)
+        return np.float64(ufunc(x, k, df))
```

Comment:
This is what I mean, something like:

```suggestion
            if df < 100000:
                cython_symbol = '_studentized_range_cdf'
                log_const = _stats._studentized_range_cdf_logconst(k, df)
                arg = [q, k, df, log_const]
                ranges = [(-np.inf, np.inf), (0, np.inf)]
            else:
                cython_symbol = '_studentized_range_cdf_asymptotic'
                arg = [q, k]
                ranges = [(-np.inf, np.inf)]
                opts = dict(epsabs=1e-11, epsrel=1e-12)
                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]

            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)

            llc = LowLevelCallable.from_cython(
                _stats, cython_symbol, usr_data)

            opts = dict(epsabs=1e-11, epsrel=1e-12)
            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
            
        ufunc = np.frompyfunc(_single_cdf, 3, 1)
        return np.float64(ufunc(x, k, df))
```